AppKit/CPWindow/CPWindow.j

```diff
@@ -1976,61 +2058,204 @@ CPTexturedBackgroundWindowMask
+                                                                                               trackingArea:aTrackingArea]];
+                                
+                                if (trackingOptions & CPTrackingCursorUpdate)
+                                    [cursorUpdateStack addObject:aTrackingArea];
+                            }
+                    }
             }
-
-            var deviation = MIN(_mouseEnteredStack.length, mouseEnteredStack.length);
-
-            while (deviation--)
-                if (_mouseEnteredStack[deviation] === mouseEnteredStack[deviation])
-                    break;
-
-            var index = deviation + 1,
-                count = _mouseEnteredStack.length;
-
-            if (index < count)
+            
+            // Search now for exited views (were in _mouseEnteredStack but no more in mouseEnteredStack)
+            
+            for (var i = 0, nb = [_mouseEnteredStack count]; i < nb; i++)
             {
-                var event = [CPEvent mouseEventWithType:CPMouseExited location:point modifierFlags:[anEvent modifierFlags] timestamp:[anEvent timestamp] windowNumber:_windowNumber context:nil eventNumber:-1 clickCount:1 pressure:0];
-
-                for (; index < count; ++index)
-                    [_mouseEnteredStack[index] mouseExited:event];
+                var aTrackingArea = [_mouseEnteredStack objectAtIndex:i];
+                
+                if (![mouseEnteredStack containsObject:aTrackingArea])
+                {
+                    // Mouse is no more in this area so it's a mouseExited
+                    
+                    if ([aTrackingArea options] & CPTrackingMouseEnteredAndExited)
+                        [[aTrackingArea owner] mouseExited:[CPEvent enterExitEventWithType:CPMouseExited
+                                                                                  location:point
+                                                                             modifierFlags:[anEvent modifierFlags]
+                                                                                 timestamp:[anEvent timestamp]
+                                                                              windowNumber:_windowNumber
+                                                                                   context:nil
+                                                                               eventNumber:-1
+                                                                              trackingArea:aTrackingArea]];
+                    
+                    // If this is the active cursor area, we reset _cursorUpdateStack so a new active area will be computed
+                    
+                    if (aTrackingArea == _activeCursorTrackingArea)
+                    {
+                        _cursorUpdateStack = [];
+                        _activeCursorTrackingArea = nil;
+                    }
+                }
```

Comment:
Do you mean between the dragging event and the movement event ?
Indeed we could put code in a method and pass some kind of "dragging:(BOOL)dragging", then add flag check but with a negative performance impact.
But this could be done if you prefer for code-styling.
Just tell me what you prefer.